scipy/spatial/_hausdorff.py

```diff
@@ -0,0 +1,54 @@
+#
+# Copyright (C)  Tyler Reddy, Richard Gowers, and Max Linke, 2016
+#
+# Distributed under the same BSD license as Scipy.
+#
+# adapted from scipy's cython version
+import numpy as np
+import random
+#pythran export directed_hausdorff(float64[:,:], float64[:,:], int)
+#pythran export directed_hausdorff(float64[:,:], float64[:,:])
+
+def directed_hausdorff(ar1, ar2, seed=0):
+
+    N1, data_dims = ar1.shape
+    N2 = ar2.shape[0]
+    i_store = j_store = i_ret = j_ret = 0
+
+    # shuffling the points in each array generally increases the likelihood of
+    # an advantageous break in the inner search loop and never decreases the
+    # performance of the algorithm
+    random.seed(seed)
+    resort1 = np.arange(N1)
+    resort2 = np.arange(N2)
+    random.shuffle(resort1)
+    random.shuffle(resort2)
+    ar1 = np.asarray(ar1)[resort1]
+    ar2 = np.asarray(ar2)[resort2]
+
+    cmax = 0
+    for i in range(N1):
+        no_break_occurred = True
+        cmin = np.inf
+        for j in range(N2):
+            d = np.sum((ar1[i] - ar2[j]) ** 2)
+            # faster performance with square of distance
+            # avoid sqrt until very end
+            if d < cmax: # break out of `for j` loop
+                no_break_occurred = False
```

Comment:
In Python `for` loops have `else` clauses which is executed if terminated normally (no `break` happened) . So you don't need to keep track of break occurance.

Reply:
Pythran supports that, so moved to that solution.